src/pages/index.js

```diff
@@ -1,41 +1,112 @@
-import React, { useContext } from "react";
-import Navbar from "../components/navbar";
-import IDE from "../components/ide";
-import Term from "../components/term";
-import Video from "../components/video";
-import { HeartwoodDispatchContext } from "../state/HeartwoodContextProvider";
+import React, { useContext, useEffect, useState } from "react"
+import useInterval from "../hooks/useInterval"
 
-import '../styles/video.css'
+import { Link } from "gatsby"
+import { HeartwoodStateContext } from "../state/HeartwoodContextProvider"
+import { isLoggedIn, signUpComplete, getEndpointPrefix } from "../utils/utils"
+import gear from "../assets/gear.svg"
 
-export default () => {
-  const dispatch = useContext(HeartwoodDispatchContext)
+import Navbar from "../components/navbar"
+import CollectInfo from "../components/collectinfo"
+
+import { format, differenceInMinutes } from "date-fns"
+
+const IndexPage = () => {
+  const state = useContext(HeartwoodStateContext)
+  const [form, setForm] = useState({ startTime: "", stopTime: "", showMeetingButton: null, lessonDescription: "", isLoading: true})
+
+  const setMeetingButton = () => {
+    const currentTime = new Date()
+    // Check if the current time is within five minutes either way of the next lesson's time
+    if (differenceInMinutes(form.startTime, currentTime) <= 5 && differenceInMinutes(currentTime, form.stopTime) <= 5) {
+      setForm({ ...form, showMeetingButton: true})
+    } else {
+      setForm({ ...form, showMeetingButton: false})
+    }
+  }
+  
+  useEffect(() => {
+    if (state.user) {
+      const endpoint = getEndpointPrefix() + "/getNextMeeting"
+      fetch(endpoint, {method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ uid: state.user.uid })})
+        .then((res) => res.json())
+        .then((res) => {
+          if (res.startTime === -1 && res.stopTime === -1) {
+            // catch case where there is not a next meeting
+            setForm({...form, 
+              lessonDescription: "You do not have any upcoming lessons scheduled", showMeetingButton: false})
+          } else {
+            // otherwise handle the case where a next meeting exists
+            const formattedStartTime = new Date(res.startTime.dateTime)
+            const formattedStopTime = new Date(res.stopTime.dateTime)
+            const lessonDescription = format(formattedStartTime,
+              "'Your next lesson is on: ' EEEE, LLLL do, y 'at' h:mm a '(PST)'")
+            setForm({...form, startTime: formattedStartTime, stopTime: formattedStopTime, lessonDescription: lessonDescription})
+            }
+        })
+        .catch((res) => {
+          console.error("unable to get next meeting")
+        })
+    }
+  }, [state.user])
```

Comment:
Codacy found an issue: [React Hook useEffect has a missing dependency: 'form'. Either include it or remove the dependency array. You can also do a functional update 'setForm(f => ...)' if you only need 'form' in the 'setForm' call.](https://app.codacy.com/gh/treetoplearning/heartwood/pullRequest?prid=5919976)